ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/ScalablePushUtil.java

```diff
@@ -82,7 +88,45 @@ public static boolean isScalablePushQuery(
         // Must be reading from "latest"
         && isLatest
         // We only handle a single sink source at the moment from a CTAS/CSAS
-        && upstreamQueries.size() == 1;
+        && upstreamQueries.size() == 1
+        // ROWPARTITION and ROWOFFSET are not currently supported in SPQs
+        && !containsPartitionOrOffset(query, ksqlConfig);
+  }
+
+  private static boolean containsPartitionOrOffset(final Query query, final KsqlConfig ksqlConfig) {
```

Comment:
Let's not bake "partition" or "offset" into any of these method names. Doing so defeats the purpose of writing the methods `isDisallowedInScalablePushQueries()` in a more generic way.